hw/ip/aes/doc/_index.md

```diff
@@ -41,7 +45,7 @@ Future versions of the AES unit might include a separate interface through which
 
 # Theory of Operations
```

Comment:
General but maybe-not-helpful comment: I think this section (and the following hardware interfaces section) is starting to have too many "conditionals" because of all of the modes.  It feels like there are a lot of "if in CBC" or "if in CTR" etc type of sentences.  I am not sure what to do about it - one option would be to describe how each mode works individually rather than try to merge them together.  That is annoying to maintain lots of parallel (mostly-duplicated) blocks of text though.  Or maybe there is some way to more consistently flag per-mode exceptions to the baseline behavior.

Sorry this is not more constructive, this problem comes up a lot with a multi-mode crypto block and is not easy to handle.  It will get even worse as you add more modes :(

Reply:
That's a valid point. I made an attempt to improve the situation a bit: The section "Datapath Architecture and Operation" is now phrased a bit more general and everything that is a peculiarity of a certain cipher mode is now in italics. This way the reader can first focus on the general operation and then dive into the different modes if needed.

I am not sure if I like the italics but I definitely prefer it over replicating and maintaining this section separately for every mode.

Alternatively, we could also split the section into a part about the cipher core only, and one about how the cipher core is fed (just like the RTL). WDYT?

Reply:
I like the italics to distinguish the mode specific sections, that was a good idea.  Let's see if anyone else has thoughts about separating the text sections - I personally don't think that you need to do that (yet).  As even more modes get added I reserve the right to change my mind ;)